src/php/Build/BuildFacade.php

```diff
@@ -41,4 +42,127 @@ public function getNamespaceFromDirectories(array $directories): array
             ->createNamespaceExtractor()
             ->getNamespacesFromDirectories($directories);
     }
+
+    /**
+     * Gets a list of all dependencies for a given list of namespaces. It first extracts all
+     * namespaces from all Phel files in the give directories and then return a
+     * topoloigcal sorted subset of these namespace information.
+     *
+     * @param string[] $directories The list of the directories
+     * @param string[] $ns A list of namespace for which we should find the subset
+     *
+     * @return NamespaceInformation[]
+     */
+    public function getDependenciesForNamespace(array $directories, array $ns): array
+    {
+        $namespaceInformation = $this->getNamespaceFromDirectories($directories);
+
+        $index = [];
+        $queue = [];
+        foreach ($namespaceInformation as $info) {
```

Comment:
We don't want to have any logic in the Facade. The Facade should be like a "proxy" delegating all logic to domain classes that the Factory creates for it. I created a [PR](https://github.com/phel-lang/phel-lang/pull/347) pointing to this one showing what do I mean by this.

The main reason is that you cannot easily write tests for the Facade. Actually, you cannot write unit tests for the Facade because it is part of the infrastructure layer (like the Factory or Config, for example).  For this reason, you want to have isolated classes with the logic of your domain, and using the Factory to create your domain classes injecting all collaborators/dependencies that your domain classes need. 

☝️ This way allows you to create independent classes with logic that is really easy to test.